manifests/init.pp

```diff
@@ -162,6 +165,16 @@
     $iftemplate = template('network/ifcfg-eth.erb')
   }
 
+  if $flush {
+    exec { 'network-flush':
+      user        => 'root',
+      command     => "/usr/sbin/ip addr flush dev ${interface}",
```

Comment:
I was going to go with "don't set a path at all", but your solution is reasonable.